docs/long_desc.rst

```diff
@@ -3,11 +3,27 @@ BanditPAM: A state-of-the-art, high-performance k-medoids algorithm
 
 Introduction
 ------------
-Clustering algorithms such as k-means are ubiquitous in modern data science applications. Despite its popularity, however, k-means has several drawbacks. K-means only supports certain distance metrics, and its cluster centers can lack interpretability. Alternatively, k-medoids uses the points in the dataset itself -- the medoids -- as cluster centers. This enables interpretability of the cluster centers. Furthermore, k-medoids supports arbitrary dissimilarity functions in place of a distance metric.
+Clustering algorithms are ubiquitous in modern data science applications. Compared to the commonly used k-means clustering, k-medoids clustering 
+uses the points in the dataset itself -- the medoids -- as cluster centers and supports arbitrary dissimilarity functions
+in place of a distance metric. This enables greater interpretability of the cluster centers and the clustering of structured 
+objects. Despite its advantages, k-medoids clustering is less popular than k-means due to its computational cost. 
+The k-medoids algorithms such as PAM and FastPAM scale quadratically in the dataset size in each iterations
+but they are still significantly slower than k-means, which scales linearly in dataset size in each iteration. We introduce BanditPAM, 
+a randomized algorithm inspired by techniques from multi-armed bandits, that runs significantly faster than prior k-medoids algorithms and achieves the same clustering results. 
 
 Details
 -------
-In this package, we provide a high-performance implementation of BanditPAM , a state-of-the-art k-medoids algorithm. BanditPAM matches state-of-the-art in clustering quality but improves the runtime of previous approaches from O(n^2) to O(nlogn) in each iteration. Furthermore, BanditPAM supports arbitrarily dissimilarity functions between points -- these functions need not even be metrics. This implementation is written in C++ for performance, but is callable from Python via Python bindings. For further details, please see the paper at https://arxiv.org/abs/2006.06856 for the full paper and the code at https://github.com/ThrunGroup/BanditPAM/blob/main/README.md
+In this package, we provide a high-performance implementation of BanditPAM, a state-of-the-art k-medoids algorithm. 
+The main contributions of BanditPAM are as follows:
+
+* BanditPAM matches the best prior work in clustering quality but scales as O(nlogn) instead of O(n^2) in each iteration.
+* BanditPAM supports arbitrary dissimilarity functions between points -- these functions need not even be metrics. 
+
```

Comment:
random newlines in line 18 and 21?

Reply:
It seems newline is needed for creating a list. I have tried without newlines in this commit a6fe82309f959d4a4f548cb57105e644e7545c0b